buildrunner/validation/step.py

```diff
@@ -0,0 +1,106 @@
+"""
+Copyright 2023 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+from typing import Any, Dict, List, Optional, Union
+
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, Field
+
+
+class StepPypiPush(BaseModel):
+    """ Step pypi push model"""
+    repository: str
+    username: str
+    password: str
+
+
+class Artifact(BaseModel):
+    """ Artifact model """
+    # Intentionally loose restrictions
+    format: Optional[str]
+    type: Optional[Any]
+    compression: Optional[str]
+    push: Optional[bool]
+
+
+class StepRun(BaseModel, extra='forbid'):
+    """ Run model within a step """
+
+    xfail: Optional[bool]
+    services: Optional[Dict[str, Any]]
```

Comment:
I feel like this should be `Optional[Dict[str, Service]]` or something so that the services get validated as well.

Reply:
It might share a lot with `run` as well, so maybe there can be a base class that is shared between them?

Reply:
I made this update, to validate the Service portion.